frontend/src/ui-components/AcmTable/AcmTable.tsx

```diff
@@ -436,13 +436,16 @@ export function AcmTable<T>(props: AcmTableProps<T>) {
       const find = columns.find((col) => col.id === id)
       find && sortedColumns.push(find!)
     })
-    // Btn column is always the last
-    const btn = columns.find((col) => col.isActionCol)
-    btn && sortedColumns.push(btn!)
 
-    return sortedColumns.filter((column) => {
+    const sortedSelected = sortedColumns.filter((column) => {
       return selectedColIds.includes(column.id as string)
     })
+
+    // Btn column is always the last
+    const btn = columns.find((col) => col.isActionCol)
+    btn && sortedSelected.push(btn!)
```

Comment:
From the SonarCloud code smell: https://sonarcloud.io/project/issues?id=open-cluster-management_console&pullRequest=2989&resolved=false&types=CODE_SMELL

The non-null assertion is unnecessary because of the leading `btn &&`

```suggestion
    btn && sortedSelected.push(btn)
```